engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -196,6 +198,7 @@
   private final List<OperatorMeta> rootOperators = new ArrayList<>();
   private final List<OperatorMeta> leafOperators = new ArrayList<>();
   private final Attribute.AttributeMap attributes = new DefaultAttributeMap();
+  private final transient List<DAGSetupPlugin> plugins = new ArrayList<>();
```

Comment:
It would be better to have this outside the LogicalPlan. One good place is in LogicalPlanConfiguration